iml-gui/crate/src/page/filesystem.rs

```diff
@@ -203,27 +290,15 @@ fn server_link<I>(uri: Option<&String>, txt: &String) -> Node<I> {
     }
 }
 
-fn volume_link<I>(cache: &ArcCache, t: &Target<TargetConfParam>) -> Node<I> {
+fn volume_link<I>(t: &Target<TargetConfParam>) -> Node<I> {
     let vol_id = match &t.volume {
-        VolumeOrResourceUri::ResourceUri(url) => extract_id(&url).unwrap().parse::<u32>().unwrap(),
+        VolumeOrResourceUri::ResourceUri(url) => extract_id(url).unwrap().parse::<u32>().unwrap(),
         VolumeOrResourceUri::Volume(v) => v.id,
     };
 
-    if let Some(vn) = cache.volume_node.arc_values().find(|v| v.volume_id == vol_id) {
-        let size = cache
-            .volume
-            .arc_values()
-            .find(|v| v.id == vol_id)
-            .map(|v| format!("({})", number_formatter::format_bytes(v.size.unwrap() as f64, None)))
-            .unwrap();
-        a![
-            class![C.whitespace_no_wrap, C.text_blue_500, C.hover__underline, C.block],
-            attrs! {At::Href => Route::Volume(RouteId::from(vol_id)).to_href()},
-            vn.path,
-            " ",
-            size
-        ]
-    } else {
-        plain!("N/A")
-    }
+    a![
+        class![C.whitespace_no_wrap, C.text_blue_500, C.hover__underline, C.block],
+        attrs! {At::Href => Route::Volume(RouteId::from(vol_id)).to_href()},
+        t.volume_name,
```

Comment:
`volume_name` is something like "360014058cba15f6bfe7408090247fd69", not a device path?

Reply:
Yes, unfortunately there is a bug where a target can point to a no longer existent volume.

However, this situation should basically never occur, you can't have a target if the backing storage doesn't exist.

This will be fixed by #1176 which uses the devices scsi id (or other unique id for virtual devices) so that even if a device is removed and re-added in the db, the pointer will still be consistent.